src/deepsparse/transformers/pipelines/nli_text_classification.py

```diff
@@ -0,0 +1,241 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# postprocessing adapted from huggingface/transformers
+
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Helper functions and pydnatic models for zero-shot text classification task
+with nli models
+"""
+
+
+from typing import TYPE_CHECKING, List, Union
+
+import numpy
+from pydantic import BaseModel, Field
+from transformers.tokenization_utils_base import PaddingStrategy, TruncationStrategy
+
+from deepsparse.utils import numpy_softmax
+
+
+if TYPE_CHECKING:
+    from deepsparse.transformers.pipelines.zero_shot_text_classification import (
+        ZeroShotTextClassificationPipeline,
+    )
+
+
+__all__ = [
+    "NliTextClassificationConfig",
+    "NliTextClassificationInput",
+    "process_nli_inputs",
+    "process_nli_engine_outputs",
+    "nli_engine_forward",
+]
+
+
+class NliTextClassificationConfig(BaseModel):
+    """
+    Schema for configuration options when running zero shot with nli
+    """
+
+    hypothesis_template: str = Field(
+        description="A formattable template for wrapping around the provided "
+        "labels to create an nli hypothesis",
+        default="This text is about {}",
+    )
+    entailment_index: int = Field(
+        description="Index of nli model outputs which denotes entailment", default=0
+    )
+    contradiction_index: int = Field(
+        description="Index of nli model outputs which denotes contradiction", default=2
+    )
+    multi_class: bool = Field(
+        description="True if class probabilities are independent", default=False
+    )
```

Comment:
i think `hypothesis_template` and `multi_class` makes more sense as a runtime input rather than a config. you might want to tweak this as you feed different labels and sequences to the pipeline

Reply:
Since there are theoretically ways of performing zero-shot text classification with hypothesis labels (like cosine similarity, detailed here: https://joeddav.github.io/blog/2020/05/29/ZSL.html) I think hypothesis label should stay nli specific.

Multiclass I definitely agree with!

Reply:
it's definitely NLI specific agreed, could it be moved to `NliTextClassificationInput`?

Reply:
I'll make it an optional input to handle both of the cases where the hypothesis template is either static and dynamic